File: src/main/java/destiny/core/chinese/FiveElementTest.java
/**
 * @author smallufo 
 * Created on 2005/7/2 at 上午 06:11:30
 */ 
package destiny.core.chinese;

import junit.framework.TestCase;

public class FiveElementTest extends TestCase
{
  public void testToString()
  {
    assertEquals(FiveElement.木.toString() , "木");
    assertEquals(FiveElement.火.toString() , "火");
    assertEquals(FiveElement.土.toString() , "土");
    assertEquals(FiveElement.金.toString() , "金");
    assertEquals(FiveElement.水.toString() , "水");
  }

  public void testGetFiveElement()
  {
    assertEquals(FiveElement.木.getFiveElement() , FiveElement.木);
    assertEquals(FiveElement.火.getFiveElement() , FiveElement.火);
    assertEquals(FiveElement.土.getFiveElement() , FiveElement.土);
    assertEquals(FiveElement.金.getFiveElement() , FiveElement.金);
    assertEquals(FiveElement.水.getFiveElement() , FiveElement.水);
  }

  public void testSame()
  {
    assertSame(FiveElement.木.getFiveElement() , FiveElement.木);
    assertSame(FiveElement.火.getFiveElement() , FiveElement.火);
    assertSame(FiveElement.土.getFiveElement() , FiveElement.土);
    assertSame(FiveElement.金.getFiveElement() , FiveElement.金);
    assertSame(FiveElement.水.getFiveElement() , FiveElement.水);

  }

  public void testGetProduct()
  {
    assertSame(FiveElement.木.getProduct() , FiveElement.火);
    assertSame(FiveElement.火.getProduct() , FiveElement.土);
    assertSame(FiveElement.土.getProduct() , FiveElement.金);
    assertSame(FiveElement.金.getProduct() , FiveElement.水);
    assertSame(FiveElement.水.getProduct() , FiveElement.木);
  }

  public void testIsProducingTo()
  {
    assertTrue(FiveElement.木.isProducingTo(FiveElement.火));
    assertTrue(FiveElement.火.isProducingTo(FiveElement.土));
    assertTrue(FiveElement.土.isProducingTo(FiveElement.金));
    assertTrue(FiveElement.金.isProducingTo(FiveElement.水));
    assertTrue(FiveElement.水.isProducingTo(FiveElement.木));
  }

  public void testGetProducer()
  {
    assertSame(FiveElement.木.getProducer() , FiveElement.水);
    assertSame(FiveElement.火.getProducer() , FiveElement.木);
    assertSame(FiveElement.土.getProducer() , FiveElement.火);
    assertSame(FiveElement.金.getProducer() , FiveElement.土);
    assertSame(FiveElement.水.getProducer() , FiveElement.金);
  }

  public void testIsProducedBy()
  {
    assertTrue(FiveElement.木.isProducedBy(FiveElement.水));
    assertTrue(FiveElement.火.isProducedBy(FiveElement.木));
    assertTrue(FiveElement.土.isProducedBy(FiveElement.火));
    assertTrue(FiveElement.金.isProducedBy(FiveElement.土));
    assertTrue(FiveElement.水.isProducedBy(FiveElement.金));
  }

  public void testGetDominateOver()
  {
    assertSame(FiveElement.木.getDominateOver() , FiveElement.土);
    assertSame(FiveElement.火.getDominateOver() , FiveElement.金);
    assertSame(FiveElement.土.getDominateOver() , FiveElement.水);
    assertSame(FiveElement.金.getDominateOver() , FiveElement.木);
    assertSame(FiveElement.水.getDominateOver() , FiveElement.火);
  }

  public void testIsDominatorOf()
  {
    assertTrue(FiveElement.木.isDominatorOf(FiveElement.土));
    assertTrue(FiveElement.火.isDominatorOf(FiveElement.金));
    assertTrue(FiveElement.土.isDominatorOf(FiveElement.水));
    assertTrue(FiveElement.金.isDominatorOf(FiveElement.木));
    assertTrue(FiveElement.水.isDominatorOf(FiveElement.火));
  }

  public void testGetDominator()
  {
    assertSame(FiveElement.木.getDominator() , FiveElement.金);
    assertSame(FiveElement.火.getDominator() , FiveElement.水);
    assertSame(FiveElement.土.getDominator() , FiveElement.木);
    assertSame(FiveElement.金.getDominator() , FiveElement.火);
    assertSame(FiveElement.水.getDominator() , FiveElement.土);
  }

  public void testIsBeatenBy()
  {
    assertTrue(FiveElement.木.isDominatedBy(FiveElement.金));
    assertTrue(FiveElement.火.isDominatedBy(FiveElement.水));
    assertTrue(FiveElement.土.isDominatedBy(FiveElement.木));
    assertTrue(FiveElement.金.isDominatedBy(FiveElement.火));
    assertTrue(FiveElement.水.isDominatedBy(FiveElement.土));
  }

}

File: src/test/java/destiny/core/chinese/FiveElementTest.kt
/**
 * @author smallufo
 * Created on 2005/7/2 at 上午 06:11:30
 */
package destiny.core.chinese

import destiny.core.chinese.FiveElement.*
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertSame
import kotlin.test.assertTrue

class FiveElementTest {
  @Test
  fun testToString() {
    assertEquals(木.toString(), "木")
    assertEquals(火.toString(), "火")
    assertEquals(土.toString(), "土")
    assertEquals(金.toString(), "金")
    assertEquals(水.toString(), "水")
  }

  @Test
  fun testGetFiveElement() {
    assertEquals(木.fiveElement, 木)
    assertEquals(火.fiveElement, 火)
    assertEquals(土.fiveElement, 土)
    assertEquals(金.fiveElement, 金)
    assertEquals(水.fiveElement, 水)
  }

  @Test
  fun testSame() {
    assertSame(木.fiveElement, 木)
    assertSame(火.fiveElement, 火)
    assertSame(土.fiveElement, 土)
    assertSame(金.fiveElement, 金)
    assertSame(水.fiveElement, 水)
  }

  @Test
  fun testGetProduct() {
    assertSame(木.product, 火)
    assertSame(火.product, 土)
    assertSame(土.product, 金)
    assertSame(金.product, 水)
    assertSame(水.product, 木)
  }

  @Test
  fun testIsProducingTo() {
    assertTrue(木.isProducingTo(火))
    assertTrue(火.isProducingTo(土))
    assertTrue(土.isProducingTo(金))
    assertTrue(金.isProducingTo(水))
    assertTrue(水.isProducingTo(木))
  }

  @Test
  fun testGetProducer() {
    assertSame(木.producer, 水)
    assertSame(火.producer, 木)
    assertSame(土.producer, 火)
    assertSame(金.producer, 土)
    assertSame(水.producer, 金)
  }

  @Test
  fun testIsProducedBy() {
    assertTrue(木.isProducedBy(水))
    assertTrue(火.isProducedBy(木))
    assertTrue(土.isProducedBy(火))
    assertTrue(金.isProducedBy(土))
    assertTrue(水.isProducedBy(金))
  }

  @Test
  fun testGetDominateOver() {
    assertSame(木.dominateOver, 土)
    assertSame(火.dominateOver, 金)
    assertSame(土.dominateOver, 水)
    assertSame(金.dominateOver, 木)
    assertSame(水.dominateOver, 火)
  }

  @Test
  fun testIsDominatorOf() {
    assertTrue(木.isDominatorOf(土))
    assertTrue(火.isDominatorOf(金))
    assertTrue(土.isDominatorOf(水))
    assertTrue(金.isDominatorOf(木))
    assertTrue(水.isDominatorOf(火))
  }

  @Test
  fun testGetDominator() {
    assertSame(木.dominator, 金)
    assertSame(火.dominator, 水)
    assertSame(土.dominator, 木)
    assertSame(金.dominator, 火)
    assertSame(水.dominator, 土)
  }

  @Test
  fun testIsBeatenBy() {
    assertTrue(木.isDominatedBy(金))
    assertTrue(火.isDominatedBy(水))
    assertTrue(土.isDominatedBy(木))
    assertTrue(金.isDominatedBy(火))
    assertTrue(水.isDominatedBy(土))
  }

}

